src/utils/mod.rs

```diff
@@ -46,6 +46,22 @@ impl PartialEq for Node {
     }
 }
 
+impl Node {
+    pub fn num_siblings(&self) -> u64 {
+        self.children.len() as u64
+    }
+
+    pub fn get_children_from_node(&self, is_reversed: bool) -> impl Iterator<Item = Node> {
+        let children = self.children.clone();
+        if is_reversed {
+            let children: Vec<Node> = children.into_iter().rev().collect();
+            return children.into_iter();
+        }
```

Comment:
if you add an 'else' on this line then we don't need the return on line 58. @baile320